docs/framework/server-side-editor-api.md

```diff
@@ -0,0 +1,264 @@
+const model = editor.data.parse( html );
+
+// Get the root element and create an insertion position.
+const root = editor.model.document.getRoot();
+const insertPosition = editor.model.createPositionAt( root, 1 );
+
+// Insert the content at the specified position.
+editor.model.insertContent( model, insertPosition );
+```
+
+This is perfect for:
+- Adding content from other places
+- Inserting ready-made templates
+- Adding pre-formatted HTML content 
+
+## Working with Track Changes
+
+{@link features/track-changes Track changes} helps you manage content suggestions effectively, for tasks like automating content review, and implementing AI-powered suggestions.
+
+### Text Modifications
+
+Let's start with a basic text replacement:
+
+```js
+// Enable track changes to mark our edits as suggestions.
+editor.execute( 'trackChanges' );
+
+// Make a simple text replacement.
+editor.execute( 'replaceAll', 'I', 'you' );
+```
+
+The `trackChanges` command ensures that all changes are marked as suggestions, making it easy for others to review and accept or reject them.
+
+### Content Structure Changes
+
+Now, let's look at how to suggest content removal:
+
+```js
+// Enable track changes to mark our edits as suggestions.
+editor.execute( 'trackChanges' );
+
+// Get the section we want to remove and prepare for deletion.
+const firstElement = editor.model.document.getRoot().getChild( 0 );
+const deleteRange = editor.model.createRangeIn( firstElement );
+const deleteSelection = editor.model.createSelection( deleteRange );
+
+// Remove the content as a suggestion.
+editor.model.deleteContent( deleteSelection );
+```
+
+This pattern is essential when building automated content review systems. You might use it to flag outdated sections, remove deprecated content, or clean up documents before publication. The API gives you precise control over what gets removed and how the changes are tracked.
+
+Similarly, you can suggest adding new content:
+
+```js
+// Enable track changes for the new content.
+editor.execute( 'trackChanges' );
+
+// Prepare the new content we want to add.
+const modelFragment = editor.data.parse( 'Hello <strong>world!</strong>' );
+
+// Add the content as a suggestion at the beginning of the document.
+const firstElement = editor.model.document.getRoot().getChild( 0 );
+const insertPosition = editor.model.createPositionAt( firstElement, 0 );
+
+editor.model.insertContent( modelFragment, insertPosition );
+```
+
+This approach shines in several real-world scenarios:
+- Automatically suggesting content updates based on external data
+- Creating templates that need review before finalization
+- Integrating with content management systems to propose changes
+- Building custom workflows for content creation and review
+
+### Attribute Modifications
+
+Sometimes you need more advanced control over how your changes are recorded, especially when working with attributes. Let's look at how to suggest attribute changes, like updating link URLs:
```

Comment:
I don't like that this description sounds like it is about some generic changes recording, while in reality we are discussing the attribute use case only.

```suggestion
Inserting and deleting content is integrated with track changes, so you can easily use `model.insertContent()` and `model.deleteContent()` as described above.

However, if you wish to create attributes suggestions using the editor model API, you need to specifically tell track changes features to record these changes. Let's look how to correctly make a suggestion to update links URLs:
```